pkg/plugin/plugin_manager.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/pkg/multicloser"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-plugin"
+)
+
+const (
+	// PluginGlob is the glob pattern used to find plugins.
+	PluginGlob = "jvs-plugin-*"
+)
+
+// PluginManager will discover and load plugins.
+type PluginManager struct {
+	// dir is the directory where plugins can be found.
+	dir string
+}
+
+// LoadPlugins loads plugins in the dir and put them into the Validator interface.
+func (p *PluginManager) LoadPlugins() (map[string]jvspb.Validator, *multicloser.Closer, error) {
+	validators := make(map[string]jvspb.Validator)
+	var merr *multierror.Error
```

Comment:
Why do we need multierror?

Reply:
With Go 1.20, we can use `errors.Join` instead of multierror.

Reply:
Thanks for the info. Updated to use erros.Join instead.